### PR TITLE
Ml/conv 1130 inconsistent notification deep link to chat

### LINKED ISSUE #1130: Bug: Following Push Notifications and Deep Links are sometimes inconsistent
### Describe the bug

Following a Push notification seems to have a 50/50 chance of going to the conversation the message was received for

This is likely happening because hydration has not finished when push notification handling is done

This means we likely need to await hydration zustand state to change

### Expected behavior

Would like to make these paths 100% successful on both iOS and Android

Case 1
- App is Closed
- User Presses Chat Notification
- User Gets Navigated to Chat for message

Case 2
- App is Backgrounded
- User Presses Chat Notification
- User Gets Navigated to Chat for message

### Steps to reproduce the bug

_No response_

### PR DESCRIPTION
**Pull request by technoplato**
Fixes #1130 

https://www.loom.com/share/3db3a0cd75524e08b9b4752e0a5784cd?sid=0bb5c799-330e-4db4-adf2-2c2160cd848d

## Release Notes

- **New Features**
	- Enhanced logging for debugging during account hydration and notification handling.
	- Streamlined navigation by adjusting how topics are managed.

- **Improvements**
	- Simplified logic for splash screen management and deep link handling.
	- Improved error handling and initialization processes for account management.

- **Bug Fixes**
	- Adjusted notification handling to prevent alerts from being shown to the sender.

### CODE REVIEW

File: components/StateHandlers/InitialStateHandler.tsx [hunk 1]
```diff
@@ -8,10 +8,13 @@ import { useSelect } from "../../data/store/storeHelpers";
 import {
   getSchemedURLFromUniversalURL,
   navigateToTopic,
+  setTopicToNavigateTo,
   topicToNavigateTo,
+  // topicToNavigateTo,
```
Comment (alexrisch) [inline components/StateHandlers/InitialStateHandler.tsx:13]: Comment  😄

File: features/notifications/utils/handleForegroundNotification.ts [hunk 2]
```diff
@@ -4,12 +4,30 @@ import * as Notifications from "expo-notifications";
...
+  logger.debug("[Notifications] Handling iOS foreground notification");
   resetNotifications();
-  const account = notification.request.content.data?.["account"];
-  if (account && account !== currentAccount()) {
+  // note(lustig): this is not the senderAddress, but the account to which
+  // the notification is to be received.
+  const recipientAccount = notification.request.content.data?.["account"];
+
+  const senderAccount =
+    /* TODO notification.request.content.data?.["sender"]; */ "todo";
```
Comment (alexrisch) [inline features/notifications/utils/handleForegroundNotification.ts:18]: Todo!

File: features/notifications/utils/onInteractWithNotification.ts [hunk 3]
```diff
@@ -51,18 +53,18 @@ export const onInteractWithNotification = (
...
-    const conversations = getChatStore(account).getState().conversations;
 
-    if (conversations[conversationTopic]) {
-      navigateToTopic(conversationTopic as ConversationTopic);
-    } else {
-      // App was probably not loaded!
-      setTopicToNavigateTo(conversationTopic);
-    }
+    navigateToTopic(conversationTopic as ConversationTopic);
+    setTopicToNavigateTo(undefined);
```
Comment (alexrisch) [inline features/notifications/utils/onInteractWithNotification.ts:67]: Is this doing anything any more?

File: components/StateHandlers/HydrationStateHandler.tsx [hunk 4]
```diff
@@ -16,14 +15,11 @@ export default function HydrationStateHandler() {
         // Awaiting before showing onboarding
         await getInstalledWallets(false);
       } else {
+        // note(lustig) I don't think this does anything?
         getInstalledWallets(false);
       }
```
[the remark this answers is not in the document]
Comment (technoplato) [reply components/StateHandlers/HydrationStateHandler.tsx:20]: this is inaccurate. there's an if else

File: features/notifications/utils/handleForegroundNotification.ts (hunk 2 above)
Comment (technoplato) [reply features/notifications/utils/handleForegroundNotification.ts:18]: I've left this half baked code in intentionally and was planning on a follow up, so wanted to leave breadcrumbs for the logic we need.

You cool leaving or want me to remove and make an Issue instead?

File: features/notifications/utils/onInteractWithNotification.ts (hunk 3 above)
Comment (technoplato) [reply features/notifications/utils/onInteractWithNotification.ts:67]: I think this lingering around was responsible for the bug where you'd come back into the app and it'd try and deep link you again, so yes!

**Review by alexrisch — APPROVED** [review]

**Pull request merged**